### PR TITLE
post migration changes

### PR DESCRIPTION
**Pull request by ramyakrishnai**
## Proposed changes
Jira: https://jira.newfold.com/browse/PRESS0-1157

Post Migration steps *****
Given when the migration is successful
When customer is SSO'ed to wp-admin dashboard
Then we will need to display the following options with -> as part of "Next Steps"
"Update your Nameserver to BH"
"Connect the site to your Domain"
And on click of the -> Help Center slider shall open with information for the following questions
How do I update my nameserver to BH?
How do I connect my site to the Domain ?

Need help with these steps ? "View guide" link to https://www.bluehost.com/help

Screenshot 2024-04-25 at 3 05 21 PM

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/configs/OnboardingList.config.js [hunk 1]
```diff
@@ -38,13 +40,45 @@ const signUpBluehostAcademy = () => {
...
+    WordPressSdk.settings.get();
+    setIsMigrationCompleted(false);
+  });
+};
 const signUpYoastSEOAcademy = () => {
   WordPressSdk.settings.put({ yoast_seo_signup_status: true });
   AnalyticsSdk.track("next_step", "next_step_yoast_academy_clicked", data);
 };
 const brandName =
   (NewfoldRuntime?.sdk?.ecommerce?.brand_settings?.name).toLowerCase();
```
Comment (aratidgr8) [inline src/configs/OnboardingList.config.js:69]: Avoid using `NewfoldRuntime?.sdk`

File: src/configs/OnboardingList.config.js [hunk 2]
```diff
@@ -57,6 +91,85 @@ export function OnboardingListDefinition(props) {
...
+        text: __(
+          "Connect this site to your domain",
+          "wp-module-ecommerce"
+        ),
+        state: {
+          isCompleted: () => check_url_match(),
+          isMigrated: (queries) => queries?.settings?.showMigrationSteps || props.isMigrationCompleted,
+        },
+
+        "data-nfdhelpcenterquery": "How do I connect my site to the Domain ?",
```
Comment (aratidgr8) [inline src/configs/OnboardingList.config.js:125]: Translations not needed?

File: src/configs/OnboardingList.config.js [hunk 3]
```diff
@@ -57,6 +91,85 @@ export function OnboardingListDefinition(props) {
...
+        state: {
+          isCompleted: (queries) => queries?.settings?.update_site_server_clicked || check_url_match(),
+          isMigrated: (queries) => queries?.settings?.showMigrationSteps || props.isMigrationCompleted
+        },
+        shouldRender: (state) => (state.isMigrated),
+        actions: {
+          manage: () => updateSiteServers(props.setWebServersUpdated),
+        },
+
+        "data-nfdhelpcenterquery": "How do I update my nameserver to BH?",
```
Comment (aratidgr8) [inline src/configs/OnboardingList.config.js:109]: Translations not needed?

File: src/configs/OnboardingList.config.js [hunk 4]
```diff
@@ -38,13 +40,45 @@ const signUpBluehostAcademy = () => {
...
+  });
+};
 const signUpYoastSEOAcademy = () => {
   WordPressSdk.settings.put({ yoast_seo_signup_status: true });
   AnalyticsSdk.track("next_step", "next_step_yoast_academy_clicked", data);
 };
 const brandName =
   (NewfoldRuntime?.sdk?.ecommerce?.brand_settings?.name).toLowerCase();
 
+const check_url_match = () => {
```
Comment (aratidgr8) [inline src/configs/OnboardingList.config.js:71]: Can we avoid redundant code of constant `check_url_match` in `src/configs/OnboardingList.config.js` and `src/components/OnboardingList.js`

Define once & import as per need.

**Comment by aratidgr8** [comment]
@ramyakrishnai Please add more details in PR description.

File: src/components/OnboardingList.js [hunk 5]
```diff
@@ -38,25 +40,40 @@ function OnboardingCheckListItem({ children, actions, state, ...props }) {
...
-          <ArrowLongRightIcon className="nfd-text-black nfd-w-[1.125rem]" />
+          (state.showText || <ArrowLongRightIcon className="nfd-text-black nfd-w-[1.125rem]" />)
         )}
       </Link>
     </li>
   );
 }
 
+const check_url_match = () => {
+  const brandName = (NewfoldRuntime?.sdk?.ecommerce?.brand_settings?.name).toLowerCase();
```
[the remark this answers is not in the document]
Comment (ramyakrishnai) [reply src/components/OnboardingList.js:66]: currently in newfoldRuntime, we're not getting ecommerce out of sdk, only once the ecommerce is out side of sdk, we can add this change

File: src/configs/OnboardingList.config.js (hunk 1 above)
Comment (ramyakrishnai) [reply src/configs/OnboardingList.config.js:69]: currently in newfoldRuntime, we're not getting ecommerce out of sdk, only once the ecommerce is out side of sdk, we can add this change

File: src/components/OnboardingList.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (aratidgr8) [reply src/components/OnboardingList.js]: Why can't we use `NewfoldRuntime?.plugin?.brand` we  can also skip `toLowerCase()` if we use that variable.

File: src/configs/OnboardingList.config.js (hunk 1 above)
Comment (aratidgr8) [reply src/configs/OnboardingList.config.js:69]: Why can't we use `NewfoldRuntime?.plugin?.brand` we  can also skip `toLowerCase()` if we use that variable.

File: src/configs/OnboardingList.config.js [hunk 6]
```diff
@@ -57,6 +91,85 @@ export function OnboardingListDefinition(props) {
...
+        text: __(
+          "Need help with these steps?",
+          "wp-module-ecommerce"
+        ),
+        state: {
+          isCompleted: (queries) => (queries?.settings?.update_site_server_clicked && check_url_match()) || check_url_match(),
+          isMigrated: (queries) => queries?.settings?.showMigrationSteps,
+          className: () => "nfd-bg-canvas",
+          hideCheck: () => true,
+          showText: () => <a className="nfd-underline" href="https://www.bluehost.com/help" target="_blank">View Guide</a>
```
[the remark this answers is not in the document]
Comment (aratidgr8) [reply src/configs/OnboardingList.config.js:142]: @ramyakrishnai plz cross-check this one as well

File: src/components/OnboardingList.js [hunk 7]
```diff
@@ -21,6 +22,7 @@ function OnboardingCheckListItem({ children, actions, state, ...props }) {
       className={classNames(
         "nfd-p-[2px]",
         "nfd-m-0 nfd-border-b nfd-border-line last:nfd-border-b-0",
+        state.className,
```
Comment (aratidgr8) [inline src/components/OnboardingList.js:25]: Can you pass `className` as props

File: src/components/OnboardingList.js [hunk 8]
```diff
@@ -38,25 +40,30 @@ function OnboardingCheckListItem({ children, actions, state, ...props }) {
...
         ) : (
-          <ArrowLongRightIcon className="nfd-text-black nfd-w-[1.125rem]" />
+          (state.showText || <ArrowLongRightIcon className="nfd-text-black nfd-w-[1.125rem]" />)
         )}
       </Link>
     </li>
   );
 }
 
+const brandName = (NewfoldRuntime?.sdk?.ecommerce?.brand_settings?.name).toLowerCase();
```
Comment (aratidgr8) [inline src/components/OnboardingList.js:65]: we use `NewfoldRuntime?.plugin?.brand` we can also skip toLowerCase() if we use that variable.

File: src/components/OnboardingScreen.js [hunk 9]
```diff
@@ -88,18 +96,18 @@ export function OnboardingScreen({
             )}
           >
             <div className="nfd-flex nfd-flex-col nfd-justify-start nfd-items-start nfd-gap-4">
-              <Title size="2">{title}</Title>
+              <Title size="2">{isMigrationCompleted ? Text.isMigrated.title : title}</Title>
```
Comment (aratidgr8) [inline src/components/OnboardingScreen.js:99]: Translations need to be added

File: src/components/OnboardingScreen.js [hunk 10]
```diff
@@ -88,18 +96,18 @@ export function OnboardingScreen({
...
               <div>
                 {comingSoon ? (
                   <Alert
                     variant="warning"
                     className="nfd-text-sm nfd-bg-transparent nfd-p-0 "
                   >
-                    <span className="nfd-text-red-700">{description}</span>
+                    <span className="nfd-text-red-700">{isMigrationCompleted ? Text.isMigrated.description : description}</span>
                   </Alert>
```
Comment (aratidgr8) [inline src/components/OnboardingScreen.js:107]: Translations need to be added

File: src/configs/OnboardingList.config.js [hunk 11]
```diff
@@ -38,13 +40,36 @@ const signUpBluehostAcademy = () => {
...
+  WordPressSdk.settings.put({ showMigrationSteps: false }).then(() => {
+    WordPressSdk.settings.get();
+    setIsMigrationCompleted(false);
+  });
+};
 const signUpYoastSEOAcademy = () => {
   WordPressSdk.settings.put({ yoast_seo_signup_status: true });
   AnalyticsSdk.track("next_step", "next_step_yoast_academy_clicked", data);
 };
 const brandName =
```
Comment (aratidgr8) [inline src/configs/OnboardingList.config.js:68]: check if `brandName` variable could also be added to constants, and imported at necessary places

File: src/components/OnboardingList.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ramyakrishnai) [reply src/components/OnboardingList.js]: We can't use plugin, becuase when bluehost changed it's brand like bluehost-us, hostgator_latam, at that time, this value wil not work

File: src/configs/OnboardingList.config.js (hunk 1 above)
Comment (ramyakrishnai) [reply src/configs/OnboardingList.config.js:69]: We can't use plugin, becuase when bluehost changed it's brand like bluehost-us, hostgator_latam, at that time, this value wil not work

File: src/components/OnboardingScreen.js (hunk 10 above)
Comment (ramyakrishnai) [reply src/components/OnboardingScreen.js:107]: translations are added in the variable which is being used here

File: src/components/OnboardingScreen.js [hunk 12]
```diff
@@ -88,18 +96,18 @@ export function OnboardingScreen({
...
                 ) : (
                   <span className="nfd-text-[--nfd-ecommerce-text-info] nfd-text-sm">
-                    {description}
+                    {isMigrationCompleted ? Text.isMigrated.description : description}
```
[the remark this answers is not in the document]
Comment (ramyakrishnai) [reply src/components/OnboardingScreen.js:110]: translations are added in the variable which is being used here

File: src/components/OnboardingScreen.js (hunk 9 above)
Comment (ramyakrishnai) [reply src/components/OnboardingScreen.js:99]: translations are added in the variable which is being used here

File: src/components/OnboardingList.js (hunk 7 above)
Comment (ramyakrishnai) [reply src/components/OnboardingList.js:25]: I'm passing classname as prop which is being destructed as state from props, these values are coming from onboarding.config.js

File: src/configs/OnboardingList.config.js (hunk 1 above)
Comment (aratidgr8) [reply src/configs/OnboardingList.config.js]: As discussed on call we would go `split` way and add a `TODO`

File: src/components/OnboardingList.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (aratidgr8) [reply src/components/OnboardingList.js]: As discussed on call we would go split way and add a TODO

File: src/configs/Utility.js [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import { NewfoldRuntime } from "../sdk/NewfoldRuntime";
+import { BH_UR_REGEX, HG_UR_REGEX } from "../constants";
+
+
+export const brandName = (NewfoldRuntime?.plugin?.brand?.split('-')[0]); // TODO: change it once runtime ecommerce is moved
```
Comment (aratidgr8) [inline src/configs/Utility.js:5]: We will also have to check for `underscore` for `hostgator_latam` isn't it?

**Review by aratidgr8 — CHANGES_REQUESTED** [review]

File: src/configs/OnboardingList.config.js [hunk 14]
```diff
@@ -57,6 +78,91 @@ export function OnboardingListDefinition(props) {
...
+        text: __(
+          "Need help with these steps?",
+          "wp-module-ecommerce"
+        ),
+        state: {
+          isCompleted: (queries) => (queries?.settings?.update_site_server_clicked && check_url_match(brandName)) || check_url_match(brandName),
+          isMigrated: (queries) => queries?.settings?.showMigrationSteps,
+          className: () => "nfd-bg-canvas",
+          hideCheck: () => true,
+          showText: () => <a className="nfd-underline" href={ NewfoldRuntime.sdk.ecommerce.brand_settings.videGuideLink } target="_blank">View Guide</a>
```
Comment (aratidgr8) [inline src/configs/OnboardingList.config.js:138]: This can be moved out of `sdk` ?

File: includes/Data/Brands.php [hunk 15]
```diff
@@ -34,7 +34,7 @@ public static function get_config( Container $container ) {
 						'woocommerce_default_country' => 'AU:NSW',
 						'woocommerce_currency'        => 'AUD',
 					),
-					'wondercartBuyNow' => ''
+					'wondercartBuyNow' => '',
```
Comment (aratidgr8) [inline includes/Data/Brands.php:37]: We were to remove all `Brands.php` changes right?

Comment (ramyakrishnai) [reply includes/Data/Brands.php:37]: yeah, the changes were removed already

**Review by aratidgr8 — APPROVED** [review]

**Comment by ramyakrishnai** [comment]
@aratidgr8 can I know why is this PR labeled as `Do not merge` even after getting your approval for this PR ?

**Review by manikantakailasa — APPROVED** [review]

**Review by manikantakailasa — APPROVED** [review]

**Pull request merged**